src/main/java/org/variantsync/diffdetective/diff/difftree/parse/DiffTreeParser.java

```diff
@@ -153,7 +165,51 @@ public static DiffTree createDiffTree(
             collapseMultipleCodeLines,
             ignoreEmptyLines,
             annotationParser
-        ).parse(fullDiff);
+        ).parse(() -> {
+            String line = fullDiff.readLine();
+            if (line == null) {
+                return null;
+            } else if (line.length() == 0) {
+                return new DiffLine(null, "");
+            } else {
+                return new DiffLine(DiffType.ofDiffLine(line), line.substring(1));
+            }
+        });
+    }
+
+    /**
+     * Parses a variation tree from a source file.
+     * This method is similar to {@link createDiffTree(BufferedReader, boolean, boolean, CPPAnnotationParser)}
+     * but acts as if all lines where unmodified.
+     *
+     * @param file The source code file (not a diff) to be parsed.
+     * @param collapseMultipleCodeLines Whether multiple consecutive artifact lines should be
+     * collapsed into a single artifact node.
+     * @param ignoreEmptyLines Whether empty lines (no matter if they are added removed or remained
+     * unchanged) should be ignored.
+     * @param annotationParser The parser to parse conditional macros.
+     * @return A parsed {@link DiffTree}.
+     * @throws IOException iff {@code file} throws an {@code IOException}
+     * @throws DiffParseException if an error in the diff or macro syntax is detected
+     */
+    public static DiffTree createVariationTree(
+            BufferedReader file,
+            boolean collapseMultipleCodeLines,
+            boolean ignoreEmptyLines,
+            CPPAnnotationParser annotationParser
+    ) throws IOException, DiffParseException {
+        return new DiffTreeParser(
+            collapseMultipleCodeLines,
+            ignoreEmptyLines,
+            annotationParser
+        ).parse(() -> {
+            String line = file.readLine();
+            if (line == null) {
+                return null;
+            } else {
+                return new DiffLine(DiffType.NON, line);
+            }
+        });
```

Comment:
The changes look very good, thank you. It would be great though if this method should throw an error when the diff contains changed lines. We can only parse variation trees from source code files and not from diffs. So even though this is a workaround anyway, it should not accept changed lines in a diff.

Reply:
This method (`createVariationTree`) doesn't receive a diff, but a source file, so it will never parse changed lines. If you provide it a diff, then it will interpret the changed lines as unchanged source lines which start with a plus or minus by chance.
Do I understand correctly that you want to check if any line starts with a plus or minus to prevent accidentally misusing `createVariationTree` for parsing diffs? This heuristic would also reject some well formed source code (although in most programming languages this would be very badly formatted code).

Reply:
Yes, preventing (accidental) misuse was my intent but I see your point that the proposed heuristic would also prevent some legal input. What about keeping the code as is but issuing a warning when a + or - is found at the beginning of a line here?

Reply:
This is probably a good idea as this will probably not often appear in practice.